From bf5b89d039e68eec21474a1b503e7ecff03c953b Mon Sep 17 00:00:00 2001
From: huangyifeng <huangyifeng@eswincomputing.com>
Date: Fri, 1 Nov 2024 10:29:40 +0800
Subject: [PATCH 105/109] WIN2030-15279:fix(bmc2.7):Bug fix for trace32 jtag
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Changelogs:
1.The MCU does not need to handle the RST_OUT signal issued by the
EIC770x; otherwise, it may cause the MCPU’s debug module to reset.
the will cause the jtag reset halt cmd fail.
2.update bmc version to 2.7

Change-Id: I84453db2116a67d95ee3e592747f94cc1ba100b6
Signed-off-by: huangyifeng <huangyifeng@eswincomputing.com>
---
 Core/Inc/hf_common.h       |  2 +-
 Core/Src/hf_gpio_process.c | 10 ++++++++--
 2 files changed, 9 insertions(+), 3 deletions(-)

diff --git a/Core/Inc/hf_common.h b/Core/Inc/hf_common.h
index c921e9e..c65bdb1 100644
--- a/Core/Inc/hf_common.h
+++ b/Core/Inc/hf_common.h
@@ -237,7 +237,7 @@ extern UART_HandleTypeDef huart6;
 #endif
 /* define ------------------------------------------------------------*/
 #define BMC_SOFTWARE_VERSION_MAJOR                   2
-#define BMC_SOFTWARE_VERSION_MINOR                   6
+#define BMC_SOFTWARE_VERSION_MINOR                   7
 
 #define MAGIC_NUMBER	0x45505EF1
 
diff --git a/Core/Src/hf_gpio_process.c b/Core/Src/hf_gpio_process.c
index 7cacaab..eb8cdd5 100644
--- a/Core/Src/hf_gpio_process.c
+++ b/Core/Src/hf_gpio_process.c
@@ -160,7 +160,7 @@ static void mcu_reset_som_process(void)
 	*/
 	es_check_carrier_board_info();
 }
-
+/*
 static void som_rst_feedback_process(void)
 {
 	bmc_debug("%s %d som reset, get_som_power_state()=%d\n",
@@ -171,7 +171,7 @@ static void som_rst_feedback_process(void)
 		som_reset_control(pdFALSE);
 	}
 }
-
+*/
 static uint8_t get_user_key_status(void)
 {
 	return HAL_GPIO_ReadPin(KEY_USER_RST_GPIO_Port, KEY_USER_RST_Pin);
@@ -232,8 +232,14 @@ void hf_gpio_task(void *parameter)
 		if (flags > 0) {
 			if (flags & FLAGS_KEY)
 				key_process();
+			/*
+			The MCU does not need to handle the RST_OUT signal issued by the EIC770x;
+			otherwise, it may cause the MCPU’s debug module to reset.
+			*/
+			/*
 			if (FLAGS_SOM_RST_OUT & flags)
 				som_rst_feedback_process();
+			*/
 			if (FLAGS_MCU_RESET_SOM & flags)
 				mcu_reset_som_process();
 			if (FLAGS_KEY_USER_RST & flags)
-- 
2.25.1

